tools/agentic_import/README.md

```diff
@@ -0,0 +1,264 @@
+
+## Setup
+
+### Step 1: Repository Setup
+
+Clone the Data Commons data repository and set up environment variables:
+
+```bash
+# Clone the repository
+git clone https://github.com/datacommonsorg/data.git
+export DC_DATA_REPO_PATH="/path/to/your/cloned/data/repo"
+cd $DC_DATA_REPO_PATH
+```
+
+### Step 2: Environment Setup
+
+Set up the Python virtual environment:
+
+```bash
+# Setup Python virtual environment
+./run_tests.sh -r
+
+# Activate the virtual environment
+source .env/bin/activate
+
+# Note: Use this Python environment for all subsequent commands
+# as it has all required dependencies installed
+```
+
+### Step 3: Working Directory Preparation
+
+Create or navigate to your working directory where input data and metadata files will be stored:
+
+```bash
+# Create working directory
+mkdir -p /path/to/working/directory
+cd /path/to/working/directory
+export WORKING_DIR=$(pwd)
+
+# Ensure you have your input data and metadata files in this directory
+# Expected structure:
+# working_directory/
+# ├── input_data.csv
+# ├── metadata_file1.json
+# ├── metadata_file2.xml
+# ├── metadata_file3.txt
+# └── ... (other metadata files)
+```
+
+> **Note**: For downloading SDMX data and metadata files, see the [SDMX CLI documentation](../sdmx_import/README.md).
+
+## Data Import Process
+
+**IMPORTANT:** Run all the following steps from within your working directory with the Python virtual environment activated.
+
+```bash
+# Confirm you are in the working directory
+cd $WORKING_DIR
+
+# Confirm virtual environment is activated
+source $DC_DATA_REPO_PATH/.env/bin/activate
+```
+
+### Step 4: Data Sampling
+
+Create a sample of your data using the data_sampler utility. This helps with initial testing and validation:
+
+```bash
+# Sample your input CSV data (maximum 30 lines)
+python $DC_DATA_REPO_PATH/tools/statvar_importer/data_sampler.py \
+  --sampler_input="$WORKING_DIR/input_data.csv" \
+  --sampler_output="$WORKING_DIR/sample_data.csv" \
+  --max_rows=30
+```
+
+**Parameters:**
+- `--sampler_input`: Path to your original CSV data file
+- `--sampler_output`: Path where the sample will be saved
+- `--max_rows`: Maximum number of rows to sample (recommended: 30 for testing)
+
+### Step 5: PV Map Generation
+
+Generate the PV map and metadata files using the pvmap_generator with command-line flags. This will use Gemini CLI to read sample data, generate pvmap.csv, metadata.csv and convert sample data to Data Commons observations as output.csv:
+
+**WARNING:** This will run Gemini CLI in YOLO mode on Linux (unrestricted access), but will use sandboxing on macOS. Be careful when running on Linux systems.
+
+**NOTE:** For troubleshooting Gemini CLI issues, refer to the [Debugging](#debugging) section.
+
+```bash
+# Generate PV map and metadata
+python $DC_DATA_REPO_PATH/tools/agentic_import/pvmap_generator.py \
+  --input_data="sample_data.csv" \
+  --input_metadata="metadata_file1.json,metadata_file2.xml,metadata_file3.txt" \
+  --sdmx_dataset
```

Comment:
Can we also take an output path prefix that can be used for all generated output files?
for example: `--output_path=output/gdp` to create files: `output/gdp_pvmap.csv, output/gdp_metadata.csv` etc